cmd/process-agent/collector.go

```diff
@@ -572,10 +584,15 @@ func (l *Collector) consumePayloads(results *api.WeightedQueue, fwd forwarder.Fo
 				responses, err = fwd.SubmitRTContainerChecks(forwarderPayload, payload.headers)
 			case checks.Connections.Name():
 				responses, err = fwd.SubmitConnectionChecks(forwarderPayload, payload.headers)
-			case checks.Pod.Name():
+			case config.PodCheckMetadataName, config.PodCheckManifestName:
```

Comment:
There is one thing that now is left after the refactoring and can be confusing:
- we are still running as the pod check 
- as soon we get to the `runCheck` part we change the checkName to manifest and metadata check and change the "name" further down the line to do different actions on the forwarder/collector

I think that can be confusing because we rely on the check name but in this case it is still the same check but different "sub-checks". Some additional documentation or even clear wrapping might help here. Let me know if that part is unclear.

Reply:
Thanks for the feedback, I made some updates here:
I decide to continue use pod check name 
https://github.com/DataDog/datadog-agent/blob/551ba56141bb206e8be1dd6021425594f156abf6/cmd/process-agent/collector.go#L180-L186
while building the payload, I use message body type instead of check name to know if it's manifest payload and to add header  https://github.com/DataDog/datadog-agent/blob/551ba56141bb206e8be1dd6021425594f156abf6/cmd/process-agent/collector.go#L281-L284

When sending pod check payloads, I use `zstd` header to decide whether send the payloads as manifest payloads
https://github.com/DataDog/datadog-agent/blob/551ba56141bb206e8be1dd6021425594f156abf6/cmd/process-agent/collector.go#L591-L599